curifactory/staging.py

```diff
@@ -707,14 +846,92 @@ def wrapper(record: Record, records: list[Record] = None, **kwargs):
             utils.set_logging_prefix("")
             record.manager.stage_active = False
             record.manager.update_map_progress(record, "continue")
+            record.stage_cachers = None
             return record
 
         return wrapper
 
     return decorator
 
 
-def _check_cached_outputs(stage_name, record, outputs, cachers, records=None):
+def _get_output_representations_for_map(
+    record, outputs: list[str], cachers, records=None
+) -> list[MapArtifactRepresentation]:
+    """Check if the requested outputs are cached but do not load them, simply return
+    a list of statuses for found."""
+    if cachers == []:

```

Comment:
Are you trying to do an empty check or type check?
Empty check should be `if len(cachers) == 0:` while type check should be `if isinstance(cachers, list):`